File: flink-test/src/main/scala/com/atguigu/wc/WordCount.scala
package com.atguigu.wc

import org.apache.flink.streaming.api.scala._

/**
  * @ObjectName WordCount
  * @Description TODO
  * @Author Xing
  * @Date 2021/4/9 20:35
  * @Version 1.0
  */
object WordCount {

  def main(args: Array[String]): Unit = {
    @volatile var bb = false
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    val lineDS: DataStream[String] = env.socketTextStream("hadoop102",9999)
    lineDS.flatMap(_.split(" "))
      .map((_,1))
      .keyBy(_._1)
      .sum(1)
      .print()
    env.execute("scala UNBOUNDED stream")
  }

}

File: flink-core/src/main/java/com/atguigu/setwc/WordCount.java
package com.atguigu.setwc;

import org.apache.flink.api.common.typeinfo.Types;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.AggregateOperator;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.operators.FlatMapOperator;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;

/**
 * @ClassName WordCount
 * @Description TODO
 * @Author Xing
 * @Date 2021/4/9 20:33
 * @Version 1.0
 */
public class WordCount {
    public static void main(String[] args) throws Exception {
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        DataSource<String> lineDS = env.readTextFile("input/words.txt");

        FlatMapOperator<String, Tuple2<String, Long>> wordAndOne = lineDS.flatMap(
                (String line, Collector<Tuple2<String, Long>> out) -> {
                    String[] strings = line.split(" ");
                    for (String word : strings) {
                        out.collect(Tuple2.of(word, 1L));
                    }
                }).returns(Types.TUPLE(Types.STRING, Types.LONG));

        AggregateOperator<Tuple2<String, Long>> sum = wordAndOne.groupBy(0).sum(1);
        sum.print();
    }
}
